### ISSUE TITLE
Inconsistent indentation for custom indentation of 2

### ISSUE THREAD

**Issue by genevieveluyt** [go-yaml/yaml#508]
Hi @niemeyer,

This is a follow-up issue for the issue my colleague opened a while back (https://github.com/go-yaml/yaml/issues/497). Thanks so much for addressing that one so quickly!

We noticed some interesting behaviour with indentation and weren't sure what the expected behaviour was. The test case is formatted the same as the referenced issue.

We are using version `gopkg.in/yaml.v3 v3.0.0-20190905181640-827449938966`

The indentation for the marshaler is set to 2 (which seems to be the only indentation level this applies to)

```
Input:
a: # comment
- b
c:
- d

Node Tree:
&{Kind:1 Style:0 Tag: Value: Anchor: Alias:<nil> Content:[0xc00037a0a0] HeadComment: LineComment: FootComment: Line:1 Column:1}
    &{Kind:4 Style:0 Tag:!!map Value: Anchor: Alias:<nil> Content:[0xc00037a140 0xc00037a1e0 0xc00037a320 0xc00037a3c0] HeadComment: LineComment: FootComment: Line:1 Column:1}
        &{Kind:8 Style:0 Tag:!!str Value:a Anchor: Alias:<nil> Content:[] HeadComment: LineComment:# comment FootComment: Line:1 Column:1}
        &{Kind:2 Style:0 Tag:!!seq Value: Anchor: Alias:<nil> Content:[0xc00037a280] HeadComment: LineComment: FootComment: Line:2 Column:1}
            &{Kind:8 Style:0 Tag:!!str Value:b Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:2 Column:3}
        &{Kind:8 Style:0 Tag:!!str Value:c Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:3 Column:1}
        &{Kind:2 Style:0 Tag:!!seq Value: Anchor: Alias:<nil> Content:[0xc00037a460] HeadComment: LineComment: FootComment: Line:4 Column:1}
            &{Kind:8 Style:0 Tag:!!str Value:d Anchor: Alias:<nil> Content:[] HeadComment: LineComment: FootComment: Line:4 Column:3}

Output:
a: # comment
  - b
c:
- d
```
Here, the presence of the comment changed the indentation of the following array. Trying out other indentation levels, the expected behaviour seems to be that arrays should be indented two spaces less than other elements at the same nesting level, so the second array in this case is indented as expected. Is that correct?

Thank you!

@genevieveluyt + @klautcomputing

**Comment by niemeyer** [comment]
Certainly looks like a bug and I'll look into it. Thanks for reporting this.

**Comment by genevieveluyt** [comment]
Hi @niemeyer,

I was just wondering if you would like any help with this or if you might know when you'll have a chance to look at this. It's a bit of a blocker for us.

Thanks!

**Comment by niemeyer** [comment]
I can reproduce it with indentation set to 2, as described. I'll fix it.

**Comment by niemeyer** [comment]
This should have been fixed by fc94e3f. Please let me know if you find any other issues.
